### ISSUE TITLE
Tips for better training

### ISSUE THREAD

**Issue by mive93** [juntang-zhuang/ShelfNet#20]
Dear @juntang-zhuang,

First of all, thank you for this repo. 
I am trying to use it to train shelfnet on the [Mapillary Vistas Dataset](https://www.mapillary.com/dataset/vistas?pKey=1GyeWFxH_NPIQwgl0onILw&lat=20&lng=0&z=1.5) ([here](https://git.hipert.unimore.it/mverucchi/shelfnet) you can find my fork). 
I have succeeded training she Real-Time version of Shelfnet, however the results are pretty bad even after 270000 epochs. 
The reached mIOU is 34,06%, however in [this paper](https://arxiv.org/pdf/1912.01394.pdf) they say that they were able to achieve 49.2% on your model. 
I have already tried to contact the authors, but I got no response.  
Therefore I wanted to ask you if you could give me some tips to improve the training to achieve better results.

Thank you in advance.

Best,
Micaela

**Comment by juntang-zhuang** [comment]
Hi, sorry I have not experimented on Mapillary, too large for me. My suggestion would be first to check your dataloader and class definition, any tiny errors there could cause a big difference. Second is you can try a smaller learning rate or a different optimizer such as Adam (it usually converges to a slightly worse result but not significantly worse). By "270000 epochs", are you referring to epochs or steps?

**Comment by mive93** [comment]
Ok thank you for your answer.
Yeah, I meant 270000 step. I think the loading is correct, but I will try with changing optimizer and lr.
I will let you know, Thanks!
